scio-core/src/main/scala/com/spotify/scio/coders/instances/AvroCoders.scala

```diff
@@ -128,11 +128,26 @@ trait AvroCoders {
 
   implicit def avroSpecificRecordCoder[T <: SpecificRecord: ClassTag]: Coder[T] = {
     val clazz = ScioUtil.classOf[T]
+
+    def getSchemaFromEmptyInstance = try {
+      clazz.getDeclaredConstructor().newInstance().getSchema
+    } catch {
+      // when T is abstract class or interface it is not possible to retrieve a schema
+      case cause: Throwable =>
+        val msg =
+          "Failed to create a coder for SpecificRecord because it is impossible to retrieve an " +
+            s"Avro schema by instantiating $clazz. Use only a concrete type implementing " +
+            s"SpecificRecord or use GenericRecord type in your transformations if a concrete " +
+            s"type is not known in compile time."
+        throw new Throwable(msg, cause)
```

Comment:
I think this isn't worth chaining the exception here, as the one raised by 
```scala
SpecificData.get().getSchema(clazz)
```
is already suppressed. We're already in a fallback/recover case.

Reply:
- my intuition is that this extra info won't hurt
- this would make clear if failure happens for anything other than "constructor not found", like `getSchema` might fail because of some bug in apache avro